sopel/cli/run.py

```diff
@@ -0,0 +1,658 @@
+        except KeyboardInterrupt:
+            break
+        except Exception:  # TODO: Be specific
+            trace = traceback.format_exc()
+            try:
+                tools.stderr(trace)
+            except Exception:  # TODO: Be specific
+                pass
+            logfile = open(os.path.join(config.core.logdir, 'exceptions.log'), 'a')
+            logfile.write('Critical exception in core')
+            logfile.write(trace)
+            logfile.write('----------------------------------------\n\n')
+            logfile.close()
+            # TODO: This should be handled by command_start
+            # All we should need here is a return value, but replacing the
+            # os._exit() call below (at the end) broke ^C.
+            # This one is much harder to test, so until that one's sorted it
+            # isn't worth the risk of trying to remove this one.
+            os.unlink(pid_file)
+            os._exit(1)
+
+        if not isinstance(delay, int):
+            break
+        if p.wantsrestart:
+            return -1
+        if p.hasquit:
+            break
+        tools.stderr(
+            'Warning: Disconnected. Reconnecting in %s seconds...' % delay)
+        time.sleep(delay)
+    # TODO: This should be handled by command_start
+    # All we should need here is a return value, but making this
+    # a return makes Sopel hang on ^C after it says "Closed!"
+    os.unlink(pid_file)
+    os._exit(0)
+
+
+def add_legacy_options(parser):
+    parser.add_argument("-d", '--fork', action="store_true",
+                        dest="daemonize", help="Daemonize Sopel")
+    parser.add_argument("-q", '--quit', action="store_true", dest="quit",
+                        help=(
+                            "Gracefully quit Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop` instead)"))
+    parser.add_argument("-k", '--kill', action="store_true", dest="kill",
+                        help=(
+                            "Kill Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel stop --kill` instead)"))
+    parser.add_argument("-r", '--restart', action="store_true", dest="restart",
+                        help=(
+                            "Restart Sopel "
+                            "(deprecated, and will be removed in Sopel 8; "
+                            "use `sopel restart` instead)"))
+    parser.add_argument("-l", '--list', action="store_true",
+                        dest="list_configs",
+                        help="List all config files found")
+    parser.add_argument("-m", '--migrate', action="store_true",
+                        dest="migrate_configs",
+                        help="Migrate config files to the new format")
+    parser.add_argument('--quiet', action="store_true", dest="quiet",
+                        help="Suppress all output")
```

Comment:
Ditto as for `--fork`/`-d`, though this could (and probably should) apply to other actions besides `start`. `# TODO`?

Reply:
I'm not sure this option is useful outside of the start command. I would rather not have it for the other sub-command instead.

Reply:
Probably any command that could be run from a script (init/service control or otherwise) should support `--quiet`. Currently that would be:

  * `start`
  * `stop`
  * `restart`

The other actions are decidedly user-facing and not useful to run from scripts, IMO. (Though I guess the Docker/Ansible guys might hackily use `sopel configure`, knowing them! :laughing:)

Reply:
Yeah, better keeping the same behavior as Sopel 6 on that one, I'll add the Output redirect in `stop` and `restart` sub-commands.

If I don't want it, I'll do another PR, or think about it more and provide better arguments later.

Reply:
Done!